pkg/discovery/pods.go

```diff
@@ -36,31 +36,42 @@ const (
 	PodLogsLocation = "podlogs"
 )
 
-// Only set values if they have values greater than 0 (as in they user specified).
-func getPodLogOptions(container string, limitBytes, sinceSeconds int64) *v1.PodLogOptions {
+// mapping to k8s.io/api/core/v1/PodLogOptions.
+func getPodLogOptions(cfg *config.Config) *v1.PodLogOptions {
```

Comment:
I think keeping the `container` in the signature is a good idea so that it is clear to any caller that that extra piece of information is needed. Otherwise you have to set it manually afterwards (which you do)

Reply:
Well...hm... I see this prevented you from regenerating the rest of the options set when iterating over the containers.

Lets just check what happens on k8s side if you leave the container empty in this; would it still be usable and default to the first container?

As long as it is usable I'm ok with it.